### PR TITLE
Expose a new Client.oauthAuthorizationURL(…) method

### PR DESCRIPTION
**Pull request by tonyarnold**
This works for my local usage, and I think it could be useful to others - but I'm still working my way through the best way to handle OAuth login on macOS. I'd welcome feedback!

```swift
let client = Client(baseURL: application.baseURL)

guard let tokenURL = client.oauthAuthorizationURL(clientID: application.clientID, scopes: application.scopes, redirectURI: application.redirectURI) else {
    // Handle failure to return a valid token URL
}

NSWorkspace.shared.open(tokenURL)

// Handle the response via your app delegate:

func application(_ application: NSApplication, open urls: [URL]) {
    for url in urls {
        if url.isOAuth2Callback {
            // Pass to Login.oauth(…)
        }
    }
}
```

### CODE REVIEW

File: Sources/MastodonKit/Client.swift [hunk 1]
```diff
@@ -60,4 +60,24 @@ public struct Client: ClientType {
...
     }
+
+    /// OAuth authorization URL. Use this to initiate a standard OAuth login.
+    ///
+    /// - Parameters:
+    ///   - clientID: The client ID.
+    ///   - scopes: The access scopes.
+    ///   - redirectURI: The client redirectURI.
+    /// - Returns: URL for the OAuth authorization endpoint.
+    public func oauthAuthorizationURL(clientID: String, scopes: [AccessScope], redirectURI: String) -> URL? {
```
Comment (marcelvoss) [inline Sources/MastodonKit/Client.swift:71]: Wouldn't this be equivalent to `Login.oauth(...)` as implemented [here](https://github.com/MastodonKit/MastodonKit/blob/master/Sources/MastodonKit/Requests/Login.swift#L49).

Comment (tonyarnold) [reply Sources/MastodonKit/Client.swift:71]: Similar, sure. This PR was from four years ago - I've no idea what or why I was trying to change here anymore.

Comment (marcelvoss) [reply Sources/MastodonKit/Client.swift:71]: Yup, I'm sorry about the delay. I've recently taken over this repository, so I'm trying to catch up with everything. If you agree, I'll go ahead and close this PR?

Thanks a lot for your contribution nonetheless!

Comment (tonyarnold) [reply Sources/MastodonKit/Client.swift:71]: > If you agree, I'll go ahead and close this PR?

Yeah, that sounds good to me. Good luck with getting the project up and running again!

**Pull request closed without merging**